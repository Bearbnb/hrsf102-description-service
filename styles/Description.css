.container {
  width: 65%;
  margin: 20px auto 10px auto;
}

.button {
  background:none!important;
  margin-top: 25px;
  color: #6BB7B9;
  border:none; 
  padding:0!important;
  font: inherit;
  cursor: pointer;
  font-size: 18px;
}

button:focus {
  outline: none;
}

button:hover {
  text-decoration: underline;
}