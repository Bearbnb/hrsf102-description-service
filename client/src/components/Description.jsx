import React from 'react';
import PropTypes from 'prop-types';

import DescriptionExtended from './DescriptionExtended.jsx';

import styles from '../../../styles/Description.css';

const Description = ({ 
  description,
  descriptionExtended,
  showDescription,
  showDescriptionExtended,
  hideDescription,
}) => {
  let expandDescriptionButton;

  if (showDescriptionExtended) {
    expandDescriptionButton = (
      <div />
    );
  } else {
    expandDescriptionButton = (
      <button
        className={styles.button}
        type="button"
        onClick={() => showDescription()}
      >Read more about the space
      </button>
    );
  }

  return (
    <div className={styles.container}>

      <p>
        {description}
      </p>

      <DescriptionExtended
        hideDescription={hideDescription}
        descriptionExtended={descriptionExtended}
        showDescriptionExtended={showDescriptionExtended}
      />
      {expandDescriptionButton}
    </div>
  );
};

Description.propTypes = {
  description: PropTypes.string,
  descriptionExtended: PropTypes.shapeOf(PropTypes.number),
  showDescription: PropTypes.func,
  hideDescription: PropTypes.func,
  showDescriptionExtended: PropTypes.bool,
};

Description.defaultProps = {
  description: '',
  descriptionExtended: {
    'THE SPACE': 'IT\'S LARGE',
    'VERY IMPORTANT:': 'DON\'T TOUCH THINGS',
    'GUEST ACCESS': 'USE THE DOOR',
  },
  showDescription: () => {},
  hideDescription: () => {},
  showDescriptionExtended: false,
};

export default Description;
