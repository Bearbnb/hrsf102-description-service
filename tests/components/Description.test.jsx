import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Description from '../../client/src/components/Description';

describe('<Description />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Description />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
