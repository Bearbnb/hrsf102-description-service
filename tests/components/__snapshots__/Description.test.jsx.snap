// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> should render properly 1`] = `
<div
  className="container"
>
  <p />
  <DescriptionExtended
    descriptionExtended={
      Object {
        "GUEST ACCESS": "USE THE DOOR",
        "THE SPACE": "IT'S LARGE",
        "VERY IMPORTANT:": "DON'T TOUCH THINGS",
      }
    }
    hideDescription={[Function]}
    showDescriptionExtended={false}
  />
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Read more about the space
  </button>
</div>
`;
